src/pytest_benchmark/plugin.py

```diff
@@ -857,7 +866,10 @@ def display_results_table(self, tr):
                 worst[prop] = max(benchmark[prop] for _, benchmark in report_progress(
                     benchmarks, tr, "{line} ({pos}/{total})", line=line))
 
-            unit, adjustment = time_unit(best.get(self.sort, benchmarks[0][self.sort]))
+            time_unit_key = self.sort
+            if self.sort in ("name", "fullname"):
```

Comment:
uhm, this might be useful, especially if you want to compare benchmarks among different runs (e.g., CPython vs PyPy), to ensure that the units are comparable. But then, I'd vote for auto:min, auto:max etc. for the automatic cases, because --benchmark-time-unit min sounds a bit too obscure, IMHO